ste/mgr-JobMgr.go

```diff
@@ -672,10 +704,72 @@ func (jm *jobMgr) CloseLog() {
 	jm.chunkStatusLogger.FlushLog()
 }
 
+// DeferredCleanupJobMgr cleanup all the jobMgr resources.
+// Warning: DeferredCleanupJobMgr should be called from JobMgrCleanup().
+//          As this function neither threadsafe nor idempotient. So if DeferredCleanupJobMgr called
+//          mulitple times, it may stuck as receiving channel already closed. Where as JobMgrCleanup()
+//          safe in that sense it will do the cleanup only once.
+//
+// TODO: Add JobsAdmin reference to each JobMgr so that in any circumstances JobsAdmin should not freed,
+//       while jobMgr running. Whereas JobsAdmin store number JobMgr running  at any time.
+//       At that point DeferredCleanupJobMgr() will delete jobMgr from jobsAdmin map.
+func (jm *jobMgr) DeferredCleanupJobMgr() {
+	jm.Log(pipeline.LogInfo, "DeferredCleanupJobMgr called")
+
+	time.Sleep(60 * time.Second)
+
+	jm.Log(pipeline.LogInfo, "DeferredCleanupJobMgr out of sleep")
+
+	// Call jm.Cancel to signal routines workdone.
+	// This will take care of any jobPartMgr release.
+	jm.Cancel()
+
+	// Cleanup the JobStatusMgr go routine.
+	jm.CleanupJobStatusMgr()
+
+	// Transfer Thread Cleanup.
+	jm.cleanupTransferRoutine()
+
+	// Remove JobPartsMgr from jobPartMgr kv.
+	jm.deleteJobPartsMgrs()
+
+	// Close chunk status logger.
+	jm.cleanupChunkStatusLogger()
+	jm.Log(pipeline.LogInfo, "DeferredCleanupJobMgr Exit, Closing the log")
+
+	// Sleep for sometime so that all go routine done with cleanUp and log the progress in job log.
+	time.Sleep(60 * time.Second)
+
+	jm.logger.CloseLog()
+}
+
 func (jm *jobMgr) ChunkStatusLogger() common.ChunkStatusLogger {
 	return jm.chunkStatusLogger
 }
 
+func (jm *jobMgr) cleanupChunkStatusLogger() {
+	jm.chunkStatusLogger.FlushLog()
+	jm.chunkStatusLogger.CloseLogger()
+}
+
+// TODO: find a better way for JobsAdmin to log (it doesn't have direct access to the job log, because it was originally designed to support multiple jobs
+func (jm *jobMgr) logJobsAdminMessages() {
+	/*
```

Comment:
Nit: Remove if comment is not required.

Reply:
This function is not getting used anywhere.

Reply:
Removed the function and comment.